util/encounter_tools.py

```diff
@@ -132,8 +132,11 @@ def choose_fight_times(args, encounters):
 
 
 # Timeline test/translate functions
-def clean_and_split_tl_line(line):
-    return re.search(r'^(?P<time>[\d\.]+)\s+"(?P<label>[^"]+)"\s+(?P<options>.+)', line.split('#')[0])
+def clean_and_split_tl_line(line, split):
+    to_parse = line
+    if split:
+        to_parse = line.split('#')[0]
```

Comment:
For my own understanding, why do you only conditionally remove the comment here?

(Also, this was wrong before your change here, but this code is probably inaccurate for things like `12.0 "Puddle #3" sync /thing/`)

Reply:
Because that specific function serves two purposes depending on that flag. It's also used in `test_timeline` to strip comments entirely, and for that purpose we specifically would want syncs to be ignored. Or at least that was the process that led me to setting it up that way. It's reasonable to re-split that function back out, I suppose.

That's a good point on in-line # symbols though. Ordinarily I would prefer to just discard that line since it's clearly an author add-in and not a game skill, but there could very well be abilities that contain the symbol. Perhaps check whether the symbol is inside quotes and then split based on that?

Reply:
Is there a need to comment out translated syncs? Really my question comes down to why is there conditional behavior? It seems like you'd want to remove comments in all cases?

Reply:
Oops, that's on me, I forgot to link the related issue when I created this request. [It's to ensure that even if there are abilities that don't get synced *in the timeline* we can still translate them.](https://github.com/quisquous/cactbot/issues/749#issuecomment-557830256) But if there's a better way to handle that situation I'd much prefer that.

Reply:
Oh right, we need to translate those so that we can find the names of the abilities in the timeline that they refer to.  Sorry for being forgetful.  (Maybe that can be a comment so that future me can remember.)

Style-wise, I'd rather have two differently named functions than one with a boolean parameter that adjusts logic.  It's just harder to follow the calling code to know what that boolean is doing, vs reading a more explanatory function name.

Reply:
That's reasonable. I think I originally combined it when we made the changes to `test_timeline`, but you're right, it is better to separate them now that other stuff needs to use that regex.